model/src/main/kotlin/OrtResult.kt

```diff
@@ -122,14 +133,22 @@ data class OrtResult(
             scanner?.results?.scanResults?.find { it.id == id }.getAllDetectedLicenses()
 
     /**
-     * Return all detected licenses associated to the projects / packages they occur in.
+     * Return all detected licenses associated to the projects / packages they occur in. If [omitExcluded] is set to
+     * true, excluded projects / packages are omitted from the result.
      */
     @Suppress("UNUSED") // This is intended to be mostly used via scripting.
-    fun collectAllDetectedLicenses() =
+    fun collectAllDetectedLicenses(omitExcluded: Boolean = false) =
             sortedMapOf<String, SortedSet<Identifier>>().also { licenses ->
+                // Note that we require the analyzer result here to determine whether a package has been implicitly
+                // excluded via its project or scope.
+                val excludes = repository.config.excludes.takeIf { omitExcluded && analyzer != null }
+
                 scanner?.results?.scanResults?.forEach { result ->
-                    result.getAllDetectedLicenses().forEach { license ->
-                        licenses.getOrPut(license) { sortedSetOf() } += result.id
+                    // At this point we know that analyzer != null if excludes != null.
+                    if (excludes?.isPackageExcluded(result.id, analyzer!!.result) == false) {
```

Comment:
You have to use `!= true` instead, because if `excludes` is `null` the above expression evaluates to `false`.